official/vision/beta/modeling/layers/nn_layers.py

```diff
@@ -720,6 +720,7 @@ def _compute_buffered_causal_padding(self,
     Returns:
       A list of paddings for `tf.pad`.
     """
+    shape_in = inputs.shape[1:-1]
```

Comment:
Let's rename to `input_shape`. Also, `inputs.shape` does not work in graph mode, so we should use `tf.shape(inputs)`:

```suggestion
    input_shape = tf.shape(inputs)[1:-1]
```

Reply:
The change to requiring the input shape will affect the `_buffered_spatial_output_shape` method. Can we change it to:

```python
  def _buffered_spatial_output_shape(self, spatial_output_shape: List[int]):
    """Computes the spatial output shape from the input shape."""
    # When buffer padding, use 'valid' padding across time. The output shape
    # across time should be the input shape minus any padding, assuming
    # the stride across time is 1.
    if self._use_buffered_input and spatial_output_shape[0] is not None:
      padding = self._compute_buffered_causal_padding(
          tf.zeros([1] + spatial_output_shape + [1]), use_buffered_input=False)
      spatial_output_shape[0] -= sum(padding[1])
    return spatial_output_shape
```